### DISCUSSION TITLE
Trigger=single spacebar function

### DISCUSSION THREAD

**Discussion by ss7box** [OpenHantek/OpenHantek6022#300] [Q&A] [answered]
Single - display a new triggered trace every time the space bar is pressed. This is from the user manual.

My experience is that single trigger works once. The spacebar turns triggering on/off when trigger icon is red. After trigger, icon is green, and spacebar acts differently - it alternately highlights the || button. The button never returns to >. No new triggers are started. The trigger will not go back to red.

If i stop/start the entire program, i can capture another event.

The probe is attached to the output of a Geiger-Müller tube board. The output is normally at 2.3V and goes low briefly when a particle is detected in the tube. 

Using 20220421 commit 1009 dso 6022BL FW0210

**Comment by Ho-Ro** [comment]
What happens when you click the `|>` / `||` button - same?
If you're on Windows you could try the key `S` instead of `SPACE` because `SPACE` is grabbed by the OS and acts differently.

**Comment by Ho-Ro** [comment, marked as answer]
Triggering on single events is imho the weakest point of the Hantek. I [explained it](https://github.com/OpenHantek/OpenHantek6022/issues/214#issuecomment-906489194) recently:
>Due to the missing trigger HW of the scope OpenHantek6022 is best suited for **repetitive** signals in the **extended audio range** (1 Hz .. 100 kHz). If you want to catch single events reliable - get another hardware.

**Comment by ss7box** [comment]
I'm on Ubuntu 20.04. Built from git clone.

The spacebar, s key, and mouse click on >|| all behave the same way.

In the Auto mode, the spacebar works as described in the manual.

In Single mode, to repeat, I exit the program. (I start the program from
command line.) I see that the program is stuck. The program will eventually
end on it's own with a message about having been in thread. (I can provide
a capture.) I usually ctl-c out to speed up the reps. It seems to me that
the problem could be user input handling in the program that runs after the
trigger is activated.

I was able to eventually capture the GM board pulse that drives a 555
monostable circuit, and the 1 ms output of that circuit using both channels
of the Hantek6022. I had to repeatedly stop/start the Openhantek program to
find the right settings and find the right RC combination to get 1 ms
output.

I'm tempted to dive into source code to see if I can activate some debug
related to Single and human I/O. I'm an EE that became a C programmer by
trade. I'm reclaiming abandoned EE interests. This little oscope and
Openhantek has pushed me along.
